third_party/airflow/armada/operators/armada.py

```diff
@@ -120,3 +127,43 @@ def _get_lookout_url(self, job_id: str) -> str:
         if self.lookout_url_template is None:
             return ""
         return self.lookout_url_template.replace("<job_id>", job_id)
+
+
+def annotate_job_request_items(
+    context, job_request_items: List[submit_pb2.JobSubmitRequestItem]
+) -> List[submit_pb2.JobSubmitRequestItem]:
+    """
+    Annotates the inbound job request items with Airflow context elements
+
+    :param context: The airflow context.
+
+    :param job_request_items: The job request items to be sent to armada
+
+    :return: annotated job request items for armada
+    """
+    task_instance = context["ti"]
+    task_id = task_instance.task_id
+    run_id = context["run_id"]
+    dag_id = context["dag"].dag_id
+
+    for item in job_request_items:
```

Comment:
My python is weak.  Does this work if annotations are not defined?  I think they are optional fields?

Reply:
i think it's an empty dictionary when empty and this should work -- would you mind running in your airflow setup to confirm? I have not set that up yet

Reply:
Sounds like a perfect time to do that.. ;)

Reply:
the sleep_job() test function provides the default annotations value in the unit test, and this works